### ISSUE TITLE
Missing or incorrect completions for numpy 1.20.1

### ISSUE THREAD

**Issue by cswartzvi** [davidhalter/jedi#1746]
I ran into a situation while using numpy 1.20.1 with vscode (jedi enabled), where some completions were either missing or had an incorrect type. I checked this with jedi.Script and it appears the problem is also present there as well. Here are some examples (using jedi  0.18.0):

-  `numpy.linspace`

```python
>>> numpy.__version__
'1.19.5'
>>> print(jedi.Script("import numpy; numpy.lins").complete())
[<Completion: linspace>]
```

```python
>>> numpy.__version__
'1.20.1'
>>> print(jedi.Script("import numpy; numpy.lins").complete())
[]
```

- `numpy.arange`

```python
>>> numpy.__version__
'1.19.5'
>>> print(jedi.Script("import numpy; numpy.arange").complete().pop().type)
function
```

```python
>>> numpy.__version__
'1.20.1'
>>> print(jedi.Script("import numpy; numpy.arange").complete().pop().type)
statement
```

I am no expert on the internals of jedi or numpy, but could this related to the new [type annotations](https://numpy.org/doc/stable/release/1.20.0-notes.html#numpy-is-now-typed) introduced in `numpy` 1.20?

**Comment by krassowski** [comment]
This would explain recent https://github.com/palantir/python-language-server/issues/906 (pining numpy to <1.20 fixes it). It also fails with jedi 0.17.2

**Comment by bnavigator** [comment]
Jedi does not see beyond the the ufunc anymore:

```pycon
>>> import jedi
>>> import numpy
>>> numpy.__version__
'1.20.1'
>>> NUMPY_DOC = '\n\nimport numpy as np\nnp.sin\n\n'
>>> pos = {'line': 4, 'column': 4}
>>> jedi.Script(NUMPY_DOC).infer(**pos)
[<Name full_name='numpy.ufunc', description='instance ufunc'>]


>>> import jedi
>>> import numpy
>>> numpy.__version__
'1.19.2'
>>> 
>>> NUMPY_DOC = '\n\nimport numpy as np\nnp.sin\n\n'
>>> pos = {'line': 4, 'column': 4}
>>> jedi.Script(NUMPY_DOC).infer(**pos)
[<Name full_name='numpy.sin', description='instance ufunc'>]
```

**Comment by cswartzvi** [comment]
It appears that my original issue concerning completions for `linspace` / `arange` has been fixed as of numpy 1.20.3 (maybe 1.20.2):

```python
>>> numpy.__version__
'1.20.3'

>>> print(jedi.Script("import numpy; numpy.lins").complete())
[<Completion: linspace>]

>>> print(jedi.Script("import numpy; numpy.ara").complete())
[<Completion: arange>]
```

Note:  @bnavigator's issue concerning jedi inference of numpy ufuncs remains open:
```python
>>> NUMPY_DOC = '\n\nimport numpy as np\nnp.sin\n\n'
>>> pos = {'line': 4, 'column': 4}
>>> jedi.Script(NUMPY_DOC).infer(**pos)
[<Name full_name='numpy.ufunc', description='instance ufunc'>]
```

@davidhalter would you prefer a separate issue for the latter?

**Comment by davidhalter** [comment]
> @davidhalter would you prefer a separate issue for the latter?

No. IMO this is not a Jedi issue. If they changed something that makes type inference worse, it's probably something that they messed up.

My guess would be that they played with decorators and Jedi cannot 100% understand it, because it's too complicated. The solution to get Jedi to understand it again is usually to add annotations.
